### PR TITLE
refactor(react-sdk/styling): ParticipantView

### PR DESCRIPTION
**Pull request by arnautov-anton**
- rename `ParticipantBox` to `ParticipantView` (to match RN)
- move UI elements (details and context menu) from `ParticipantView` to it's own files, newly consume through `ParticipantViewUI` prop (as a component)
- remove styling from the "primitive" component `BaseVideo` and move it to `Video`
- remove redundant styling (extra video wrapper, `--wide` class...)
- apply `ParticipantView` changes to layout presets

TODO:
- [x] check styling in sample applications
- [x] apply `ParticipantView` changes to `egress-composite`

### CODE REVIEW

File: packages/react-sdk/src/core/components/ParticipantView/DefaultParticipantViewUI.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                className={clsx(
+                  'str-video__participant-details__connection-quality',
+                  `str-video__participant-details__connection-quality--${connectionQualityAsString}`,
+                )}
+                title={connectionQualityAsString}
+              />
+            )}
+          </Notification>
+        )}
+        {indicatorsVisible && !hasAudio && (
```
Comment (MartinCupela) [inline packages/react-sdk/src/core/components/ParticipantView/DefaultParticipantViewUI.tsx:134]: @arnautov-anton I am thinking, whether this should be handled via CSS and class name. Maybe somebody will want to keep the indicator there, just change the icon.

File: packages/react-sdk/src/core/components/ParticipantView/ParticipantView.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  (
+    {
+      participant,
+      sinkId,
+      videoKind = 'video',
+      muteAudio,
+      setVideoElementRef,
+      className,
+      VideoPlaceholder,
+      ParticipantViewUI,
```
Comment (MartinCupela) [inline packages/react-sdk/src/core/components/ParticipantView/ParticipantView.tsx:69]: @arnautov-anton So we will keep the pattern of custom component props?

File: packages/react-sdk/src/core/components/ParticipantView/DefaultParticipantViewUI.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  showMenuButton?: boolean;
+} & Pick<ParticipantViewProps, 'participant'>;
+
+const ToggleButton = forwardRef<HTMLButtonElement, ToggleMenuButtonProps>(
+  (props, ref) => {
+    return <IconButton enabled={props.menuShown} icon="ellipsis" ref={ref} />;
+  },
+);
+
+export const DefaultParticipantViewUI = ({
```
Comment (MartinCupela) [inline packages/react-sdk/src/core/components/ParticipantView/DefaultParticipantViewUI.tsx:44]: 👍 for prepending the component name with `Default`

File: packages/react-sdk/src/core/components/ParticipantView/ParticipantView.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+   * - `<video />` for participants with enabled video.
+   * - `<div />` for participants with disabled video. This ref would point to
+   * the VideoPlaceholder component.
+   *
+   * @param element the element ref.
+   */
+  setVideoElementRef?: (element: HTMLElement | null) => void;
+
+  /**
+   * An additional list of class names to append to the root DOM element.
```
Comment (MartinCupela) [inline packages/react-sdk/src/core/components/ParticipantView/ParticipantView.tsx:54]:
```suggestion
   * Custom class applied to the root DOM element.
```

File: packages/react-sdk/src/core/components/ParticipantView/ParticipantView.tsx (hunk 2 above)
Comment (arnautov-anton) [reply packages/react-sdk/src/core/components/ParticipantView/ParticipantView.tsx:69]: For core, mostly logical components, yes.

File: packages/react-sdk/src/core/components/Video/BaseVideo.tsx [hunk 5]
```diff
@@ -1,38 +1,36 @@
...
-  VideoHTMLAttributes,
 } from 'react';
-import clsx from 'clsx';
 import { Browsers } from '@stream-io/video-client';
 
-export type VideoProps = DetailedHTMLProps<
-  VideoHTMLAttributes<HTMLVideoElement>,
-  HTMLVideoElement
-> & {
+export const applyElementRef = <T extends HTMLElement | null>(
```
Comment (MartinCupela) [inline packages/react-sdk/src/core/components/Video/BaseVideo.tsx:10]: 👍

File: packages/react-sdk/src/core/components/ParticipantView/DefaultParticipantViewUI.tsx (hunk 1 above)
Comment (arnautov-anton) [reply packages/react-sdk/src/core/components/ParticipantView/DefaultParticipantViewUI.tsx:134]: I'm going to keep it as is for now as this is UI component which we should encourage users to build themselves. This PR aimed at this separation from the `ParticipantView(Box)`.

File: packages/react-sdk/src/core/components/Video/BaseVideo.tsx (hunk 5 above)
Comment (arnautov-anton) [reply packages/react-sdk/src/core/components/Video/BaseVideo.tsx:10]: Ah, I should probably move this to utilities.

File: packages/styling/src/ParticipantView/ParticipantView-layout.scss [hunk 6]
```diff
@@ -1,22 +1,30 @@
...
   max-width: 1280px;
-  min-height: 0;
   max-height: 100%;
   aspect-ratio: 4/3;
 
-  .str-video__participant_details,
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border-radius: var(--str-video__border-radius-sm);
```
Comment (MartinCupela) [inline packages/styling/src/ParticipantView/ParticipantView-layout.scss:14]: Probably should go to theme file.

File: packages/styling/src/ParticipantView/ParticipantView-layout.scss [hunk 7]
```diff
@@ -1,22 +1,30 @@
...
+
+  &--speaking {
+    outline: 2px solid var(--str-video__primary-color);
```
Comment (MartinCupela) [inline packages/styling/src/ParticipantView/ParticipantView-layout.scss:17]: Probably should go to theme file.

File: packages/styling/src/CallParticipantsScreenView.scss [hunk 8]
```diff
@@ -88,19 +88,7 @@
     }
   }
```
[the remark this answers is not in the document]
Comment (arnautov-anton) [reply packages/styling/src/CallParticipantsScreenView.scss:89]: I'll do this in a separate PR as this one is getting stale and is already pretty big for what it is.

File: packages/react-sdk/src/core/components/ParticipantView/DefaultParticipantViewUI.tsx [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+                className={clsx(
+                  'str-video__participant-details__connection-quality',
+                  `str-video__participant-details__connection-quality--${connectionQualityAsString}`,
+                )}
+                title={connectionQualityAsString}
+              />
+            )}
+          </Notification>
+        )}
+        {indicatorsVisible && !hasAudio && (
```
Comment (MartinCupela) [inline packages/react-sdk/src/core/components/ParticipantView/DefaultParticipantViewUI.tsx:134]:
```suggestion
{/* FIXME: handle visibility by applying CSS to a class */}
        {indicatorsVisible && !hasAudio && (
```

**Review by MartinCupela — APPROVED** [review]

**Pull request merged**